accepted/single-file/staging.md

```diff
@@ -0,0 +1,139 @@
+# Single-file Staging 
+
+Publishing apps as a single file is a popular feature-request in .Net Core. Ideally, we want a single-file solution that:
+
+* Is compatible with all .Net Core applications
+* Bundles MSIL, R2R, native code and custom (data) files
+* Doesn't require installation or cleanup steps
+* Runs directly from the bundle, without extracting components to disk
+* Reduces publish-size
+* Improves startup cost
+* Works cohesively with debuggers, profilers, Watson dump etc. 
+
+This document explores a few options to realize the single-file publish feature, with different feature-set vs development cost trade-offs. The document can also be considered a development staging plan, where each stage spills fewer items onto temporary files, while paying an incremental development cost.
+
+## 1. Self-Extractor
+
+The first stage is to develop a pack and extract tool.
+While this stage is technically simplistic, the interface and tooling will match the final solution, giving partner teams and potential customers a chance to prepare for adopting the technology.
+
+### 1.1 Description
+
+This stage implements:
+
+* A  packaging tool (bundler) that embeds the application and all of its dependencies (essentially the contents of the publish directory) into a host executable. 
+	* This version will not support compression of the bundled assemblies.
+* When the executable runs, it will extract all of those files into a temporary directory, and then run as though the app was published to that temporary directory.
+
+The mechanism will support:
+
+* Reuse: Extract on first-run, reuse on subsequent runs.
+* Upgrade: Each version of the app extracts to a unique location, supporting side-by-side use of multiple versions.
+* Uninstall: Users can identify and delete extracted files when the app is no longer needed.
```

Comment:
In extract.md, the temporary extraction was explained to clean up the temporary location on exit, so how does this work with this explanation where there the extracted files are reused?

And how exactly would users be able to *identify* no longer needed files? Especially when running an application through multiple versions, I would imagine that it would be quite difficult for users to identify which version they are running and which files they can delete (especially since this would still be a manual process).

Reply:
Removal of persistent-extracted files will be manual in this version, as noted [here](https://github.com/dotnet/designs/pull/52/files#diff-5e93215a94e626495c10448a40fa4ccbR64). That is, user can look for the app in `$HOME/.dotnet/Apps/<AppName>` and delete the apps he doesn't want to use anymore.

Here, versioning is based on a hash-of the entire single-file (including assemblies, binary and data files, signatures etc). That is, if the app, or any of its dependencies changed when recompiled, we will extract to a new location. So, it is tricky to provide an accurate human-readable description generically for all apps.

However, we can provide some information to help this situation. For example, as part of the end-marker, we can write out the full name and assembly version of the app's main assembly (or version of all assemblies in the app).